### PR TITLE
Deal with cleaned up vars.json on investigation tab

### PR DESCRIPTION
**Pull request by perlpunk**
If vars.json doesn't exist (any more), don't try to open it and leave
an error in the log.

Issue: https://progress.opensuse.org/issues/110392

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -1866,26 +1866,31 @@ sub investigate ($self, %args) {
...
-            $dir = needledir($self->DISTRI, $self->VERSION);
-            my $refspec_needles_range = "$before->{NEEDLES_GIT_HASH}..$after->{NEEDLES_GIT_HASH}";
-            $inv{needles_log} = $self->git_log_diff($dir, $refspec_needles_range);
-            $inv{needles_log} ||= 'No needle changes recorded, test regression due to needles unlikely';
-            $inv{needles_diff_stat} = $self->git_diff($dir, $refspec_needles_range) if $inv{needles_log};
+        my ($prev_file, $self_file) = map {
+            eval { Mojo::File->new($_->result_dir(), 'vars.json')->slurp }
+              // undef
+        } ($prev, $self);
+        if ($prev_file) {
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/Jobs.pm:1873]: `last unless $prev_file;` would avoid having to indent all the code for this case.

**Review by Martchus — APPROVED** [review]

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Schema/Result/Jobs.pm:1873]: Also I think that at least the line `$inv{diff_packages_to_last_good} = $self->packages_diff($prev, $ignore) // 'Diff of packages not available';` is not relying on being able to read the files. I suggest to move that line up and then use

```suggestion
        last unless $prev_file;
```

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (perlpunk) [reply lib/OpenQA/Schema/Result/Jobs.pm]: done

**Review by okurz — APPROVED** [review]

**Comment by kalikiana** [comment]
FTR [OBS: openQA - openSUSE_Leap_15.3/x86_64](https://build.opensuse.org/package/show/devel:openQA:GitHub:os-autoinst:openQA:PR-4628/openQA) failed in `t/03-auth.t` and I triggered a rebuild.

**Review by Martchus — APPROVED** [review]

**Pull request merged**